.evergreen/clone-mongo-c-driver.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+set -o xtrace
+set -o errexit
+
+MONOGO_C_DRIVER_REF=1.17.0
+
+# Force checkout with lf endings since .sh must have lf, not crlf on Windows
+git clone git@github.com:mongodb/mongo-c-driver.git --config core.eol=lf --config core.autocrlf=false
```

Comment:
If the ref will always be a tag or a branch name, it will be better to check out the branch directly with `--branch=${ref} --depth=1`, unless we need the entire repo history for some reason. It depends on if the mongo-c-driver actually takes a long time to clone or not.

(Unless this is also using a very very very old Git version that doesn't know how to do `--branch` clone. That has bitten me in the past, and even better that it would just issue a warning and clone from remote `HEAD` as if nothing was wrong. That was quite a few hours of wasted time. The repo history in question was so absurdly large that it was faster to build a newer `git` from source on-the-fly as part of the build so that I could do the shallow-clone. This anecdote is almost entirely irrelevant, but I felt like sharing here.)

Reply:
Oh good point, done.

Oh that is good to know. The last [patch build](https://evergreen.mongodb.com/version/614a14a20ae60674e6363b01) succeeded. I checked Ubuntu 14.04 and RHEL 6 from spawn hosts. Both have new enough git to support clone with `--branch`.